src/stores/console.js

```diff
@@ -0,0 +1,72 @@
+'use strict';
+
+const alt = require('../alt');
+const { clearOutput, output  } = require('../actions/console');
+const ConsoleBuffer = require('../../plugins/editor/console-buffer');
+
+class ConsoleStore {
+  constructor() {
+
+    this.bindListeners({
+      onClearOutput: clearOutput,
+      onOutput: output
+    });
+
+    this.state = {
+      buffer: new ConsoleBuffer(),
+      lastRefresh: 0,
+      refreshDelayMillis: 64,
+      refreshQueued: null
+    };
+
+  }
+
+  onClearOutput() {
+    const { buffer, refreshQueued } = this.state;
+    buffer.clear();
+    if(refreshQueued != null){
+      clearInterval(refreshQueued);
+      this.setState({ refreshQueued: null });
+    }
+    this.setState({ lastRefresh: 0 });
+  }
+
+  onOutput(evt) {
+    const { buffer, lastRefresh, refreshDelayMillis, refreshQueued } = this.state;
+
+    let allowEmit = false;
+
+    buffer.update(evt);
+
+    const refreshBuffer = function() {
+      this.setState({
+        lastRefresh: Date.now(),
+        refreshQueued: null
+      });
+      allowEmit = true;
+    }.bind(this);
+
+    if(refreshQueued != null){
+      return false;
```

Comment:
seems weird that you need to explicitly return `false` because alt doesn't emit a change if you haven't called `setState`